crates/ruff_server/src/session/settings.rs

```diff
@@ -70,6 +70,10 @@ pub(crate) struct ClientSettings {
     exclude: Option<Vec<String>>,
     line_length: Option<LineLength>,
     configuration_preference: Option<ConfigurationPreference>,
+    // These settings are only needed for tracing, and are only read from the global configuration.
+    // These will not be in the resolved settings.
+    log_level: Option<crate::trace::LogLevel>,
+    log_file: Option<PathBuf>,
 }
```

Comment:
How about creating a `GlobalSettings` struct that includes a `client_settings` fields attributed with `#[flatten]`?